shell/components/AppModal.vue

```diff
@@ -150,19 +151,7 @@ export default defineComponent({
         };
       }
 
-      if (!props.triggerFocusTrapWatcherBased) {
-        useBasicSetupFocusTrap('#modal-container-element', opts);
-      }
-    }
-  },
-  created() {
-    // This usecase is to cover the PromptModal scenario where it's renders a generic component inside.
-    // Due to the architecture of the PromptModal it needs to be handled with a watcher based focus trap
-    // but with a dedicated unmount hook
-    if (this.triggerFocusTrapWatcherBased) {
-      const opts:any = DEFAULT_FOCUS_TRAP_OPTS;
-
-      useWatcherBasedSetupFocusTrapWithDestroyIncluded(() => this.focusTrapWatcherBasedVariable, '#modal-container-element', opts, true);
+      useWatcherBasedSetupFocusTrapWithDestroyIncluded(() => props.focusTrapWatcherBasedVariable ?? props.autoTriggerFocusTrapWatcher, '#modal-container-element', opts, true);
```

Comment:
Thanks for updating this, but I think I'm still missing a detail about how these props are expected to work with each other. Here's what I see:

1. `focusTrapWatcherBasedVariable` is of type `boolean?` with a default value of `undefined`
2. `autoTriggerFocusTrapWatcher` is of type `boolean` with a default value of `true`
3. When `focusTrapWatcherBasedVariable` is `null` or `undefined`, we use the value of `autoTriggerFocusTrapWatcher` (defaults to `true`)

With the above being true, can this behavior be controlled with a single prop via `focusTrapWatcherBasedVariable ?? true`? What is the purpose of having two props in this scenario?

Reply:
The `focusTrap` composable method has been reduced to a watcher-based trigger, as per https://github.com/rancher/dashboard/blob/master/shell/composables/focusTrap.ts#L53

Afaik we can't just pass a "value" to the watcher. Only variables, hence why we can't just pass `true` https://vuejs.org/guide/essentials/watchers.html